dio/lib/src/adapters/io_adapter.dart

```diff
@@ -115,15 +115,21 @@ class IOHttpClientAdapter implements HttpClientAdapter {
         if (v != null) request.headers.set(k, v);
       });
     } on SocketException catch (e) {
-      if (!e.message.contains('timed out')) {
-        rethrow;
+      if (e.message.contains('timed out')) {
+        throw DioException.connectionTimeout(
+          requestOptions: options,
+          timeout: options.connectTimeout ??
+              httpClient.connectionTimeout ??
+              Duration.zero,
+          error: e,
+          stackTrace: StackTrace.current,
+        );
       }
-      throw DioException.connectionTimeout(
+      throw DioException.connectionError(
         requestOptions: options,
-        timeout: options.connectTimeout ??
-            httpClient.connectionTimeout ??
-            Duration.zero,
+        reason: e.message,
         error: e,
+        stackTrace: StackTrace.current,
```

Comment:
`StackTrace.current` is wrong, either we take the original stacktrace by catching it. Or none which will lead to a usable default.

Reply:
Yea, I understand, sorry.
If the others persist in having the original stacktrace, then using the catched stacktrace is correct.